fdbclient/MasterProxyInterface.h

```diff
@@ -431,6 +435,52 @@ struct GetDDMetricsRequest {
   }
 };
 
+struct DebugReadTxnStateStoreReply {
+	constexpr static FileIdentifier file_identifier = 3220576;
+	Standalone<RangeResultRef> kvs;
+	Version version;
+
+	explicit DebugReadTxnStateStoreReply() = default;
+	explicit DebugReadTxnStateStoreReply(const Standalone<RangeResultRef>& kvs, Version version)
+	  : kvs(kvs), version(version) {}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, kvs, version);
+	}
+};
+
+struct DebugReadTxnStateStoreRequest {
+	constexpr static FileIdentifier file_identifier = 4663403;
+	int limit;
+	ReplyPromise<DebugReadTxnStateStoreReply> reply;
+
+	const KeyRange& getKeys() const {
+		validate();
+		return keys;
+	}
+
+	explicit DebugReadTxnStateStoreRequest() = default;
+	explicit DebugReadTxnStateStoreRequest(const KeyRange& keys, int limit) : keys(keys), limit(limit) { validate(); }
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, keys, limit, reply);
+		if (Ar::isDeserializing) {
+			validate();
+		}
+	}
+
+private:
+	void validate() const {
+		ASSERT(KeyRangeRef(LiteralStringRef("\xff/TESTONLYtxnStateStore/"),
+		                   LiteralStringRef("\xff/TESTONLYtxnStateStore0"))
```

Comment:
If I hadn't defined a key range for this in SystemData, it seems good to do so.

Reply:
I think I avoided using it here since otherwise fdbclient would depend on fdbserver

Reply:
SystemData is in fdbclient? https://github.com/apple/foundationdb/blob/master/fdbclient/SystemData.h

Reply:
Apparently including SystemData.h breaks the build, I'll figure out what to do about it tomorrow